### PR TITLE
Landing pages Home

### PR DESCRIPTION
**Pull request by chu4**
Added landing Layout for home Controller
with new header + used existing session new View for the filling of the page
image

### CODE REVIEW

File: app/assets/stylesheets/landing.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.home-back-image{
+  color: rgba(255,255,255,0.6);
+  background: linear-gradient(rgba(0,0,0,.5), rgba(0,0,0,.5)),url('https://www.kenresearch.com/blog/wp-content/uploads/2017/11/Jordan-telecommunications-market.jpg');
```
Comment (egorodessa) [inline app/assets/stylesheets/landing.scss]: download pic, at least for speed and in case it will be not available

Comment (igorkasyanchuk) [reply app/assets/stylesheets/landing.scss]: of course download. ALL resources hosted locally in your app.

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 class HomeController < ApplicationController
+  layout "landing"
```
Comment (igorkasyanchuk) [inline app/controllers/home_controller.rb]: this is good

Comment (igorkasyanchuk) [reply app/controllers/home_controller.rb]: + create an empty pages for Pricing, etc
Just empty page with `h1 Pricing` for example

File: app/views/layouts/_header.html.slim [hunk 3]
```diff
@@ -1,7 +1,7 @@
 nav.header-navbar.navbar-expand-md.navbar.navbar-with-menu.navbar-static-top.navbar-dark.navbar-border
   .navbar-wrapper
     .navbar-container.content
-      = link_to 'SalesAssistant', new_session_path(resource_name), id: 'logo'
+      = link_to 'SalesAssistant', "#", id: 'logo'
```
Comment (igorkasyanchuk) [inline app/views/layouts/_header.html.slim]: що за прикол з лінками які нікуди не ведуть?
лого завжди веде на головну сторінку, таке правило

File: app/views/layouts/_header.html.slim [hunk 4]
```diff
@@ -44,11 +44,11 @@ nav.header-navbar.navbar-expand-md.navbar.navbar-with-menu.navbar-static-top.nav
...
-      - else
-        - if controller_name != 'sessions'
-          .collapse.navbar-collapse.float-right
-            .nav.navbar-nav.float-right
-              #login.nav-item
-                = link_to new_session_path(resource_name), class:"btn btn-success btn-md" do
-                  i.fa.fa-user
-                  | Log in
\ No newline at end of file
+      /- else
```
Comment (igorkasyanchuk) [inline app/views/layouts/_header.html.slim]: а це кому лишив комент?

File: app/views/layouts/landing.html.slim [hunk 5]
```diff
@@ -0,0 +1,13 @@
+doctype html
+html
+  head
+    title
+      | HeroesTeamApp
+    = csrf_meta_tags
+    = stylesheet_link_tag    'application', media: 'all'
+    = javascript_include_tag 'application'
+  body
+    = render 'layouts/header_landing'
```
Comment (igorkasyanchuk) [inline app/views/layouts/landing.html.slim]: перенеси цей паршал в /shared/header

File: app/views/layouts/landing.html.slim [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    title
+      | HeroesTeamApp
+    = csrf_meta_tags
+    = stylesheet_link_tag    'application', media: 'all'
+    = javascript_include_tag 'application'
+  body
+    = render 'layouts/header_landing'
+    = yield
+    = any_login_here if defined?(AnyLogin)
+    = render 'layouts/footer'
```
Comment (igorkasyanchuk) [inline app/views/layouts/landing.html.slim]: і футер теж

File: app/views/layouts/_header.html.slim (hunk 3 above)
Comment (chu4) [reply app/views/layouts/_header.html.slim]: Потрібна головна сторінка після логування, якись аккаунт чи юзер пейдж. Був прописаний рут який зараз сторінка лендінгу

**Comment by chu4** [comment]
Changed to a light header/footer + different local background image
Added root_path for authenticated user
image

File: app/views/layouts/_header.html.slim (hunk 3 above)
Comment (igorkasyanchuk) [reply app/views/layouts/_header.html.slim]: це мало би бути дашборд (чи той ПР ще не замеоджаний?)

**Comment by chu4** [comment]
New container look
image

File: app/controllers/home_controller.rb [hunk 7]
```diff
@@ -1,3 +1,11 @@
...
+
   def index; end
+
+  def about_us; end
+
+  def pricing; end
+
+  def contacts; end
```
Comment (Kmalynss) [inline app/controllers/home_controller.rb]: Do we still need this empty actions? AFAIK there is no need in them if there are templates with appropriate names.

Comment (igorkasyanchuk) [reply app/controllers/home_controller.rb]: very simple templates must be added
with h1 Pricing for example

Comment (Kmalynss) [reply app/controllers/home_controller.rb]: I understand. I'm saying that we don't need those definitions if we have templates. As i remember.

Comment (chu4) [reply app/controllers/home_controller.rb]: Yes works without actions in the controller

**Comment by chu4** [comment]
New version
image
image

**Comment by chu4** [comment]
-Resolved width bug
-Added Raleway font
-Changed the main background image as negotiated with the team today
-Refactored text to the suggested example
image
image

**Comment by chu4** [comment]
add landing page accecable for authenticated user with different header(dashbooard + logout)
find better font family
think about header

**Comment by chu4** [comment]
**CHECK LIST**
- [ ] specs added
- [ ] specs passed
- [ ] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [ ] PR is reviewed manually again (to make sure you have 100% ready code)
- [ ] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs
- [ ] screenshot of rails_best_practices and other tools added

screenshot from 2018-01-07 17-11-39

- [ ] screenshot of feature is added if task has UI

not signed_in
screenshot from 2018-01-07 17-24-50
screenshot from 2018-01-07 17-24-59

signed_in
screenshot from 2018-01-07 17-26-57
screenshot from 2018-01-07 17-27-02

**Comment by igorkasyanchuk** [comment]
For landing layout do not show header with dashboard, companies, etc. Just show everything as for not logged in user. Basically everything is the same except sign up and sign in links are not visible.

**Comment by chu4** [comment]
Should i just add log out instead of sign in&up?

**Comment by igorkasyanchuk** [comment]
What you have now is OK. But where is the of the links in header. Pricing, contact, etc.

**Comment by chu4** [comment]
for signed in user i moved those to footer. so that landing links wont mess with pages available for signed in user

File: app/views/layouts/shared/header/_header_landing.html.slim [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      li.nav-item
+        = link_to "HOME", home_path, class:"nav-link font-medium-2"
+      li.nav-item
+        = link_to "ABOUT US", about_us_path, class:"nav-link font-medium-2"
+      li.nav-item
+        = link_to "PRICING", pricing_path, class:"nav-link font-medium-2"
+      li.nav-item
+        = link_to "CONTACTS", contacts_path, class:"nav-link font-medium-2"
+  - if user_signed_in?
+    .horizontal-menu.navbar-container
```
Comment (igorkasyanchuk) [inline app/views/layouts/shared/header/_header_landing.html.slim]: don't we have such code already in other layout?
extract to partial.
this dropdown when user is logged in.
Also "arrow" next to username in this layout looks not aligned well.

Comment (chu4) [reply app/views/layouts/shared/header/_header_landing.html.slim]: edited the arrow and made a partial for that bit of a header
screenshot from 2018-01-08 22-21-30

File: app/assets/stylesheets/css/bootstrap-extended.css [hunk 9]
```diff
@@ -490,8 +490,8 @@ Bootstrap Enhance */
 .dropdown-toggle:focus {
   outline: 0; }
 
-.navbar-horizontal .dropdown-toggle::after {
-  top: -9px !important; }
+/*.navbar-horizontal .dropdown-toggle::after {*/
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/css/bootstrap-extended.css]: how this change affected admin layout? did you test it?

Comment (chu4) [reply app/assets/stylesheets/css/bootstrap-extended.css]: this style made arrow not be alined

Comment (chu4) [reply app/assets/stylesheets/css/bootstrap-extended.css]: application layout header is not affected with that change
screenshot from 2018-01-09 13-13-59

**Comment by chu4** [comment]
last comment is a possible fix for production. Bug where header didn't render properly

File: app/assets/stylesheets/css/bootstrap-extended.css (hunk 9 above)
Comment (igorkasyanchuk) [reply app/assets/stylesheets/css/bootstrap-extended.css]: why you keep this comment in CSS? do you need it? IF NOT - remove

File: app/assets/stylesheets/landing.scss [hunk 10]
```diff
@@ -0,0 +1,56 @@
+.land-font{
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/landing.scss]: add space before "{"

File: app/assets/stylesheets/landing.scss [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  text-align: center;
+}
+
+.landing-cont-back2{
+  margin: 0;
+  background: rgba(93, 255, 245, 0.13);
+  text-align: center;
+}
+
+.landing-cont-back3{
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/landing.scss]: and here ... and in other places

File: app/views/layouts/shared/header/_header_landing.html.slim [hunk 12]
```diff
@@ -0,0 +1,27 @@
+.header-navbar.navbar-expand-sm.navbar.navbar-horizontal.navbar-light.navbar-shadow.border-grey.border-lighten-2.navbar-brand-center.space-between.land-font.fixed-top.pl-5 data-nav="brand-center"
+  /! Horizontal menu content
+  .horizontal-menu.navbar-container.ml-5
+    = link_to root_path do
+      = image_tag("logo1.png", alt: 'Sales Assistant', id: 'logo')
+  .horizontal-menu.navbar-container.ml-0
+    ul.nav.navbar-nav data-menu="menu-navigation"
+      li.nav-item
+        = link_to "HOME", home_path, class:"nav-link font-medium-2"
```
Comment (igorkasyanchuk) [inline app/views/layouts/shared/header/_header_landing.html.slim]: for upper case use CSS for navigation in main layout

**Comment by chu4** [comment]
image
image

**Pull request merged**